internal/pkg/cli/app_status.go

```diff
@@ -0,0 +1,446 @@
+	if err := o.configSvc(); err != nil {
+		return err
+	}
+	serviceArn, err := o.describer.GetServiceArn(o.envName)
+	if err != nil {
+		return fmt.Errorf("get service ARN: %w", err)
+	}
+	clusterName, err := serviceArn.ClusterName()
+	if err != nil {
+		return fmt.Errorf("get cluster name: %w", err)
+	}
+	serviceName, err := serviceArn.ServiceName()
+	if err != nil {
+		return fmt.Errorf("get service name: %w", err)
+	}
+	service, err := o.ecsSvc.Service(clusterName, serviceName)
+	if err != nil {
+		return fmt.Errorf("get ECS service %s: %w", serviceName, err)
+	}
+	tasks, err := o.ecsSvc.ServiceTasks(clusterName, serviceName)
+	if err != nil {
+		return fmt.Errorf("get ECS tasks for service %s: %w", serviceName, err)
+	}
+	var taskStatus []ecs.TaskStatus
+	for _, task := range tasks {
+		status, err := task.TaskStatus()
+		if err != nil {
+			return fmt.Errorf("get status for task %s: %w", *task.TaskArn, err)
+		}
+		taskStatus = append(taskStatus, *status)
+	}
+	metrics, err := o.cwSvc.GetAlarmsWithTags(map[string]string{
+		stack.ProjectTagKey: o.ProjectName(),
+		stack.EnvTagKey:     o.envName,
+		stack.AppTagKey:     o.appName,
+	})
+	if err != nil {
+		return fmt.Errorf("get CloudWatch alarms: %w", err)
+	}
+	appStatus := webAppStatus{
+		Service: service.ServiceStatus(),
+		Tasks:   taskStatus,
+		Metrics: metrics,
+	}
+	if o.shouldOutputJSON {
+		data, err := appStatus.JSONString()
+		if err != nil {
+			return err
+		}
+		fmt.Fprintf(o.w, data)
+	} else {
+		fmt.Fprintf(o.w, appStatus.HumanString())
+	}
+
+	return nil
+}
+
+func (o *appStatusOpts) askProject() error {
+	if o.ProjectName() != "" {
+		return nil
+	}
+	projNames, err := o.retrieveProjects()
+	if err != nil {
+		return err
+	}
+	if len(projNames) == 0 {
+		return fmt.Errorf("no project found: run %s please", color.HighlightCode("project init"))
+	}
+	proj, err := o.prompt.SelectOne(
+		appStatusProjectNamePrompt,
+		appStatusProjectNameHelpPrompt,
+		projNames,
+	)
+	if err != nil {
+		return fmt.Errorf("select projects: %w", err)
+	}
+	o.projectName = proj
+
+	return nil
+}
+
+func (o *appStatusOpts) retrieveProjects() ([]string, error) {
+	projs, err := o.storeSvc.ListProjects()
+	if err != nil {
+		return nil, fmt.Errorf("list projects: %w", err)
+	}
+	projNames := make([]string, len(projs))
+	for ind, proj := range projs {
+		projNames[ind] = proj.Name
+	}
+	return projNames, nil
+}
+
+func (o *appStatusOpts) askAppEnvName() error {
+	var appNames []string
+	var envs []*archer.Environment
+	var err error
+	if o.appName == "" {
+		appNames, err = o.retrieveAllAppNames()
+		if err != nil {
+			return err
+		}
+		if len(appNames) == 0 {
+			return fmt.Errorf("no applications found in project %s", color.HighlightUserInput(o.ProjectName()))
+		}
+	} else {
+		app, err := o.storeSvc.GetApplication(o.ProjectName(), o.appName)
+		if err != nil {
+			return fmt.Errorf("get application: %w", err)
+		}
+		appNames = []string{app.Name}
+	}
```

Comment:
why do we need this? can we not have default: `appNames = []string{o.appName}`

Reply:
that's a good question. I don't know why I wrote this garbage

Reply:
Oh that is because I didn't do the validation for if app exists

Reply:
lol harsh

Reply:
It's my fault - my change to move app selector to its own struct is still sitting on my laptop... Maybe i'll work on that today.